src/main.ts

```diff
@@ -82,6 +94,21 @@ const getUpdateStatus = async (uuid: string): Promise<any> => {
 	}
 };
 
+const checkLock = async (): Promise<boolean> => {
+	return lockfile.check(lockFile)
+		.then((isLocked: boolean) => {
+			if (isLocked) {
+				return true;
+			} else {
+				return false;
+			}
+		})
+		.catch((e: any) => {
+			console.error(e.message);
+			return false;
+		});
+};
```

Comment:
This should either actually use async/await
```suggestion
const checkLock = async (): Promise<boolean> => {
	try {
		const isLocked = await lockfile.check(lockFile)
		if (isLocked) {
			return true;
		} else {
			return false;
		}
	} catch(e: any) {
		console.error(e.message);
		return false;
	}
};
```
or not declare itself as async
```suggestion
const checkLock = (): Promise<boolean> => {
	return lockfile.check(lockFile)
		.then((isLocked: boolean) => {
			if (isLocked) {
				return true;
			} else {
				return false;
			}
		})
		.catch((e: any) => {
			console.error(e.message);
			return false;
		});
};
```

Fwiw I'd say actually using async/await would be the way to go as it has many benefits